Zombie.java

```diff
@@ -1,17 +1,35 @@
+import greenfoot.*;
+
 public class Zombie extends SmoothMover {
 	
 	public int strength;
 	public int resistance;
 	public int speed;
 	
 	public Zombie() {
-		//TODO: image is temporarily
-		setImage("./ppl2.png");
+		this(10, 10, 10);		
 	}
 	
 	public Zombie(int strength, int resistance, int speed) {
+		//TODO: image is temporarily
+		setImage("./ppl2.png");
 		this.strength = strength;
 		this.resistance = resistance;
 		this.speed = speed;
+		dropCurrency(this);
+		attackGate();
```

Comment:
~~dropCurrency(this);~~
~~attackGate();~~

The method calls probably are not so well-placed in the constructor ;)
if you already want to add them, please define an act method (through override). ^^